### PR TITLE
Adding Pinchot library to interface with JoeScan JS-50 Scan Heads.

### PR DESCRIPTION
**Pull request by joeynelson**

### CODE REVIEW

File: P/Pinchot/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/pinchot-c-api
+mkdir build
+cd build/
+cmake .. -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON
+make
+make install
+install_license ${WORKSPACE}/srcdir/pinchot-c-api/LICENSE.txt
```
Comment (giordano) [inline P/Pinchot/build_tarballs.jl:21]:
```suggestion
make -j${nproc}
make install
install_license ../LICENSE.txt
```

File: P/Pinchot/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline P/Pinchot/build_tarballs.jl:26]: Let's try to build for all platforms, including experimental ones:
```suggestion
platforms = supported_platforms(; experimental=true)
```

File: P/Pinchot/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+products = [
+    LibraryProduct("libpinchot", :libpinchot)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"11.1.0")
```
Comment (giordano) [inline P/Pinchot/build_tarballs.jl:39]: We generally try to use the oldest GCC version possible, using latest one can cause big compatibility problems (and it's the reason why the job is failing).  I'd recommend removing this option and see if this works:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

**Review by giordano — COMMENTED** [review]
Thanks for your contribution!

**Comment by joeynelson** [comment]
@giordano Thanks for the assistance.

**Comment by giordano** [comment]
Thanks again to you for your PR!

**Pull request merged**